### PR TITLE
Collection of small fixes for the web api

### LINKED ISSUE #5607: Final touches on monitoring API
Final monitoring API bug fixes for beta1 readiness:

- [x] Pipeline stats from `/_node/stats/pipeline` should also be included in the parent `/_node/stats` resource for completeness; verbosity shouldn’t be an issue as this API should be for computers not humans
- [x] `/_node/stats/mem` resource should be removed and the object should instead be nested under `/_node/stats/jvm` along with the ‘thread’ stats
- [x] `/_node/stats/events` object+resource should be removed as these metrics now instead reside under the pipeline stats `/_node/stats/pipeline`
- [x] There is a bug in `/_node/stats/pipeline` where the plugin specific stats should be keyed by “plugins” and not “pipeline"
- [x] Adding input plugin specific stats, like what we have for filters/outputs already
- [x] `/_node/hot_threads?human=true` human option needs to be fixed, currently returns in JSON
- [x] Enabling specification of a comma-separated list of resources https://github.com/elastic/logstash/issues/5609

/cc @jordansissel @suyograo @andrewvc

### PR DESCRIPTION
**Pull request by purbon**
List of errors fixed:

- [x] Pipeline stats from `/_node/stats/pipeline` should also be included in the parent `/_node/stats` resource for completeness; verbosity shouldn’t be an issue as this API should be for computers not humans
- [x] `/_node/stats/mem` resource should be removed and the object should instead be nested under `/_node/stats/jvm` along with the ‘thread’ stats
- [x] `/_node/stats/events` object+resource should be removed as these metrics now instead reside under the pipeline stats `/_node/stats/pipeline`
- [x] There is a bug in `/_node/stats/pipeline` where the plugin specific stats should be keyed by “plugins” and not “pipeline"

Fixes #5607

### CODE REVIEW

File: logstash-core/spec/api/spec_helper.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # encoding: utf-8
 API_ROOT = File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "lib", "logstash", "api"))
 
+require "stud/task"
```
Comment (andrewvc) [inline logstash-core/spec/api/spec_helper.rb]: Why was this added?

**Comment by andrewvc** [comment]
Aside from the stud/task comment LGTM. Once we close that one out this is 👍

File: logstash-core/spec/api/spec_helper.rb (hunk 1 above)
Comment (purbon) [inline logstash-core/spec/api/spec_helper.rb]: because if you try to run only the api test with `bin/rspec` without this in the helper it does not work.

**Comment by purbon** [comment]
@andrewvc what do you think?

File: logstash-core/spec/api/spec_helper.rb (hunk 1 above)
Comment (andrewvc) [inline logstash-core/spec/api/spec_helper.rb]: What code is using this? I don't see any code in the PR using stud. 

Is there a missing require at the top of a .rb file somewhere? We probably shouldn't put requires in the spec_helper for libraries used by app code.

Comment (purbon) [inline logstash-core/spec/api/spec_helper.rb]: code using is https://github.com/elastic/logstash/blob/master/logstash-core/spec/api/spec_helper.rb#L59, i only added this because I run the test with `bin/rspec` only for the api, not for everything. if I don't add the requirement here it does not work.

Comment (andrewvc) [inline logstash-core/spec/api/spec_helper.rb]: Makes sense

**Comment by andrewvc** [comment]
LGTM
